### ISSUE TITLE
feature gate

### ISSUE THREAD

**Issue by till** [k0sproject/k0s#5309]
### Is your feature request related to a problem? Please describe.

I expected `PodResourcesResize` to be available in 1.31.2 — but it does not seem to be enabled.

Is enabling this via k0sctl/k0s config a non invasive operation?

In addition, any thoughts why the feature seems off. It should have graduated to GA with 1.30?

Here's my environment:

```
root@ca1303f6deb7:/workspace# kubectl explain pod.spec.containers.resources
KIND:       Pod
VERSION:    v1

FIELD: resources <ResourceRequirements>


DESCRIPTION:
    Compute Resources required by this container. Cannot be updated. More info:
    https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
    ResourceRequirements describes the compute resource requirements.
    
FIELDS:
  claims	<[]ResourceClaim>
    Claims lists the names of resources, defined in spec.resourceClaims, that
    are used by this container.
    
    This is an alpha field and requires enabling the DynamicResourceAllocation
    feature gate.
    
    This field is immutable. It can only be set for containers.

  limits	<map[string]Quantity>
    Limits describes the maximum amount of compute resources allowed. More info:
    https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

  requests	<map[string]Quantity>
    Requests describes the minimum amount of compute resources required. If
    Requests is omitted for a container, it defaults to Limits if that is
    explicitly specified, otherwise to an implementation-defined value. Requests
    cannot exceed Limits. More info:
    https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
```

I expected `resizePolicy` in the output above.

### Describe the solution you would like

Ideally a list of features enabled by default in the k0s releases.

### Additional context

```
root@ca1303f6deb7:/workspace# kubectl version
Client Version: v1.31.2
Kustomize Version: v5.4.2
Server Version: v1.31.2+k0s
```

**Comment by twz123** [comment]
I did a quick internet search and wasn't able to find that feature gate. However, I found [`InPlacePodVerticalScaling`](https://kubernetes.io/docs/concepts/workloads/autoscaling/#requirements-for-in-place-resizing), which is still alpha.

K0s doesn't do anything special in regard to feature gates, it should behave just as upstream Kubernetes. Enabling feature gates should be [straight forward](https://docs.k0sproject.io/v1.31.2+k0s.0/configuration/#specfeaturegates), though.
